pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -200,11 +205,101 @@ func (r *ReconcileMemberStatus) memberOperatorHandleStatus(reqLogger logr.Logger
 	return err
 }
 
+// loadCurrentResourceUsage loads the current usage of the cluster and stores it into the member status
+func (r *ReconcileMemberStatus) loadCurrentResourceUsage(reqLogger logr.Logger, memberStatus *toolchainv1alpha1.MemberStatus) error {
+	memberStatus.Status.ResourceUsage.MemoryUsagePerNodeRole = map[string]int{}
+	allocatableValues, err := r.getAllocatableValues(reqLogger)
+	if err != nil {
+		return err
+	}
+
+	nodeMetricsList := &v1beta1.NodeMetricsList{}
+	if err := r.client.List(context.TODO(), nodeMetricsList); err != nil {
+		return err
+	}
+
+	usagePerRole := map[string]float32{}
+	allocatablePerRole := map[string]float32{}
+NodeMetricsList:
+	for _, nodeMetric := range nodeMetricsList.Items {
+		for key, usage := range nodeMetric.Usage {
+			if key == "memory" {
+				nodeInfo, found := allocatableValues[nodeMetric.Name]
+				if found {
+					usagePerRole[nodeInfo.role] += float32(usage.Value())
+					allocatablePerRole[nodeInfo.role] += float32(nodeInfo.allocatable.Value())
+
+					delete(allocatableValues, nodeMetric.Name)
+				} else {
+					reqLogger.Info("skipping NodeMetrics resource - there wasn't found corresponding node that would be monitored", "name", nodeMetric.Name)
+				}
+				continue NodeMetricsList
+			}
+		}
+		return fmt.Errorf("memory item not found in NodeMetrics: %v", nodeMetric)
+	}
+
+	if len(allocatableValues) > 0 {
```

Comment:
could you add a comment to explain what you're doing here (for the sake of maintenance)

Reply:
also, maybe rename `allocatableValues` to `pendingNodes` to explain that it's a list from which you remove entries as you get the `memory` usage from the metrics?

Reply:
I added a comment here: e9a2517
the `pendingNodes` doesn't really explain what the map is really used for (from my point of view)

Reply:
ok, fair enough. A comment to explain the purpose is good, too.